rest-api/etl/etl.sql

```diff
@@ -0,0 +1,2062 @@
+-- -------------------------------------------------------------------
+-- @2015-2017, Odysseus Data Services, Inc. All rights reserved
+-- PPI OMOP CDM Conversion
+-- last updated September 13, 2017
+--
+-- script for ETL workflow, consisting of:
+--
+-- populating table cdm.src_clean (and aux tables)
+-- populating table cdm.src_mapped (and aux tables)
+--
+-- populating table cdm.location
+-- populating table cdm.person
+-- populating table cdm.observation
+-- populating table cdm.procedure_occurrence
+--
+-- post-conversion deduplication of event tables
+--
+-- populating table cdm_target.location
+-- populating table cdm_target.person
+-- populating table cdm_target.observation
+-- populating table cdm_target.procedure_occurrence
+-- -------------------------------------------------------------------
+
+-- -------------------------------------------------------------------
+-- table: tmp_clean_all
+-- -------------------------------------------------------------------
+
+DROP TABLE IF EXISTS cdm.tmp_clean_all;
+CREATE TABLE cdm.tmp_clean_all
+(
+    participant_id              bigint,
+    date_of_survey              datetime,
+    question_ppi_code           varchar(200),
+    question_code_id            bigint,
+    value_ppi_code              varchar(200),
+    topic_value                 varchar(200),
+    value_code_id               bigint,
+    value_number                decimal(20,6),
+    value_boolean               tinyint,
+    value_date                  datetime,
+    value_string                varchar(1024),
+    questionnaire_response_id   bigint,
+    unit_id                     varchar(50)
+);
+
+-- -------------------------------------------------------------------
+-- Rules all together
+-- -------------------------------------------------------------------
+
+INSERT INTO cdm.tmp_clean_all
+SELECT
+    pa.participant_id               AS participant_id,
+    qr.created                      AS date_of_survey,
+    co_q.value                      AS question_ppi_code,
```

Comment:
We now have a short_value field on the code table, which will be populated in cases where value > 50 characters. We should do something like IFNULL(co_q.short_value, co_q.value) so that short_value is used when set.

Reply:
Dan, thank you for your comments!
Probably it will be good to populate short_value field with all the codes, both short and cut? Then we would just replace "value" with "short_value" everywhere. The reference "value" is being used in qa.sql frequently, replacing "value" with "IFNULL(..)" will make the code less readable. And, likely, slower.

Reply:
OK. I can look into doing that. Please update this to use short_value.